operator/internal/handlers/internal/rules/rules.go

```diff
@@ -0,0 +1,132 @@
+package rules
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	lokiv1beta1 "github.com/grafana/loki/operator/api/v1beta1"
+	"github.com/grafana/loki/operator/internal/external/k8s"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ListAlertingRules returns a k8s resource list of AlertingRules for the given spec or an error. Three cases apply:
+// - Return only matching rules in the stack namespace if no namespace selector given.
+// - Return only matching rules in the stack namespace and in namespaces matching the namespace selector.
+// - Return no rules if rules selector does not apply at all.
+func ListAlertingRules(ctx context.Context, k k8s.Client, stackNs string, rs *lokiv1beta1.RulesSpec) (lokiv1beta1.AlertingRuleList, error) {
+	nsl, err := selectRulesNamespaces(ctx, k, stackNs, rs)
+	if err != nil {
+		return lokiv1beta1.AlertingRuleList{}, err
+	}
+
+	rules, err := selectAlertingRules(ctx, k, rs)
+	if err != nil {
+		return lokiv1beta1.AlertingRuleList{}, err
+	}
+
+	var lrl lokiv1beta1.AlertingRuleList
+	for _, rule := range rules.Items {
+		for _, ns := range nsl.Items {
+			if rule.Namespace == ns.Name {
+				lrl.Items = append(lrl.Items, rule)
+				break
+			}
+		}
+	}
+
+	return lrl, nil
+}
+
+// ListRecordingRules returns a k8s resource list of AlertingRules for the given spec or an error. Three cases apply:
+// - Return only matching rules in the stack namespace if no namespace selector given.
+// - Return only matching rules in the stack namespace and in namespaces matching the namespace selector.
+// - Return no rules if rules selector does not apply at all.
+func ListRecordingRules(ctx context.Context, k k8s.Client, stackNs string, rs *lokiv1beta1.RulesSpec) (lokiv1beta1.RecordingRuleList, error) {
```

Comment:
Do we really need two functions for this? I think a single `ListRules()` which returns both alerting and recording rules might be all that is needed.